composer/algorithms/weight_standardization/weight_standardization.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import textwrap
+
+import torch
+import torch.nn.utils.parametrize as parametrize
+from torch import nn
+from torch.fx import symbolic_trace
+
+from composer.core import Algorithm, Event, State
+from composer.loggers import Logger
+from composer.utils import module_surgery
+
+log = logging.getLogger(__name__)
+
+__all__ = ['apply_weight_standardization', 'WeightStandardization']
+
+
+def _standardize_weights(W: torch.Tensor):
+    """Function to standardize the input weight ``W``"""
+    reduce_dims = list(range(1, W.dim()))
+    W_var, W_mean = torch.var_mean(W, dim=reduce_dims, keepdim=True, unbiased=False)
+    return (W - W_mean) / (torch.sqrt(W_var + 1e-10))
+
+
+class WeightStandardizer(nn.Module):
+    """Class used to apply weight standardization with torch's parametrization package."""
+
+    def forward(self, W):
+        return _standardize_weights(W)
+
+
+def apply_weight_standardization(module: torch.nn.Module, n_last_layers_ignore: int = 0):
+    """`Weight Standardization <https://arxiv.org/abs/1903.10520>`_ standardizes convolutional weights in a model.
+
+    Args:
+        module (torch.nn.Module): the torch module whose convolutional weights will be parametrized.
+        n_last_layers_ignore (int, optional): the number of layers at the end of the module to not apply weight standardization.
+            Default: ``0``.
+    """
+    modules_to_parametrize = (nn.Conv1d, nn.Conv2d, nn.Conv3d)
+    # Attempt to symbolically trace a module, so the results of .modules() will be in the order of execution
+    try:
+        module_trace = symbolic_trace(module)
```

Comment:
Do you know why this might fail? If so, maybe include a bit about it in the warning.

Reply:
This was actually failing with our BERT model, but I did not dig into it 😅

Reply:
It looks like this was failing due to unsupported operations in the forward function. Specifically, there is a for-loop with an if-statement in the `forward()` of our huggingface integration which is not support by `symbolic_trace()`. I added more details to the warning. Let me know if it is sufficient!